src/services/completions.ts

```diff
@@ -3459,13 +3460,13 @@ function getCompletionData(
             }
         }
 
-        if (!isTypeLocation) {
-            // GH#39946. Pulling on the type of a node inside of a function with a contextual `this` parameter can result in a circularity
-            // if the `node` is part of the exprssion of a `yield` or `return`. This circularity doesn't exist at compile time because
-            // we will check (and cache) the type of `this` *before* checking the type of the node.
-            typeChecker.tryGetThisTypeAt(node, /*includeGlobalThis*/ false);
+        // GH#39946. Pulling on the type of a node inside of a function with a contextual `this` parameter can result in a circularity
+        // if the `node` is part of the exprssion of a `yield` or `return`. This circularity doesn't exist at compile time because
+        // we will check (and cache) the type of `this` *before* checking the type of the node.
+        typeChecker.tryGetThisTypeAt(node, /*includeGlobalThis*/ false);
+        let type = typeChecker.getTypeAtLocation(node).getNonOptionalType();

```

Comment:
On second look here, is this going to end up being accidentally slow? The old code here only pulled at this if `!isTypeLocation`, but the new code always does it. Would it make more sense to only do this if one of the other two branches is going to be executed?

Reply:
> Would it make more sense to only do this if one of the other two branches is going to be executed?

Sure, I pushed out this requested change.